pkg/kademlia/dialer.go

```diff
@@ -47,29 +48,37 @@ func (dialer *Dialer) Close() error {
 }
 
 // Lookup queries ask about find, and also sends information about self.
-func (dialer *Dialer) Lookup(ctx context.Context, self pb.Node, ask pb.Node, find pb.Node) (_ []*pb.Node, err error) {
+// If self is nil, pingback will be false.
```

Comment:
When will self be nil?

Reply:
perhaps having "self" indicate whether we want ping backs is confusing, but most kademlia lookups don't need ping backs, and therefore, don't need to send self along in the request. there are only a few kademlia lookups that do need pingbacks, and those are specifically kademlia-driven bucket refreshes. having pingbacks enabled for other lookups (pkg/discovery, etc), is too resource intensive